qiskit/aqua/operators/weighted_pauli_operator.py

```diff
@@ -748,23 +756,24 @@ def evaluate_with_result(self, result, statevector_mode, use_simulator_operator_
             raise AquaError("Operator is empty, check the operator.")
 
         avg, std_dev, variance = 0.0, 0.0, 0.0
-        if statevector_mode:
-            if use_simulator_operator_mode:
-                temp = \
-                    result.data(
-                        circuit_name_prefix + 'aer_mode')[
-                            'snapshots']['expectation_value']['test'][0]['value']
-                avg = temp[0] + 1j * temp[1]
-            else:
-                quantum_state = np.asarray(result.get_statevector(circuit_name_prefix + 'psi'))
-                for weight, pauli in self._paulis:
-                    # all I
-                    if np.all(np.logical_not(pauli.z)) and np.all(np.logical_not(pauli.x)):
-                        avg += weight
-                    else:
-                        quantum_state_i = \
-                            result.get_statevector(circuit_name_prefix + pauli.to_label())
-                        avg += (weight * (np.vdot(quantum_state, quantum_state_i)))
+        if use_simulator_operator_mode:
+            expval_data = result.data(
+                circuit_name_prefix + 'aer_mode')['snapshots']['expectation_value']['expval'][0]
+            expval = expval_data['value']
+            var = expval_data.get('variance', [0.0, 0.0])
```

Comment:
the meaning of variance here is different from the variance we compute when decomposing Hamiltonian into Pauli basis, I think we should leave std_dev is 0 for now.